Adds docker compose examples

Spoffy opened the pull request:
This adds two example docker-compose files:
* A basic Grist instance backed by sqlite, with no additional services.
* A Grist instance that uses Postgres, Redis and MinIO.

These give a starting point for people to create their own compose files, and to try Grist out of the box.

However, neither is suitable for a production/public environment, and both files are commented with that accordingly.

paulfitz commented on docker-compose-examples/grist-local-testing.yml:
```diff
@@ -0,0 +1,17 @@
+    image: gristlabs/grist:latest
+    environment:
+      # Use Python 3 instead of 2.
+      PYTHON_VERSION: 3
```
Oh, it that needed? `PYTHON_VERSION_ON_CREATION` at least is set to 3 I believe. Is something Python 2 ish happening by default?

Spoffy replied:
Turns out it isn't, I copied it from an old example. I see we now have this in the docker image by default anyway :)

jordigh commented:
Generally, I like this. Could we parametrise some of the obvious variables like password or persistent storage with [interpolation](https://docs.docker.com/compose/environment-variables/set-environment-variables/#additional-information)?

paulfitz commented on docker-compose-examples/grist-with-postgres-redis-minio.yml:
```diff
@@ -0,0 +1,88 @@
+      # Redis setup
+      REDIS_URL: redis://grist-redis
+
+      # MinIO setup
```
Could maybe have a comment about how the bucket needs to have version control enabled (I see you have code to do this later)

Spoffy replied:
Good shout, done!

paulfitz commented on docker-compose-examples/grist-with-postgres-redis-minio.yml:
```diff
@@ -0,0 +1,88 @@
+# - MinIO for snapshot storage
+
+# It is STRONGLY RECOMMENDED not to use this container in a way that makes it accessible to the internet.
+# This setup lacks basic security or authentication.
```
Auth is something people really want, probably ranking over external db / redis / snapshots.

This is certainly a useful example though!

Spoffy replied:
Added an Auth example with OIDC. :)

Spoffy commented:
Thanks for the reviews! Adding the OIDC example, then I'll incorporate the feedback from you both :)

Spoffy commented:
Added some interpolation-based config variables @jordigh, added more docs and also moved all the volumes to exist in a single "persist" sub-directory to make it easier to wipe them all.

jordigh commented on docker-compose-examples/grist-traefik-oidc-auth/README.md:
```diff
@@ -0,0 +1,24 @@
+This example can then be run with `docker compose up`. This will make Grist available on `https://grist.localhost` with a self-signed certificate (by default), after all the services have started. Note: it may take up to a minute for all of the services to start correctly.
+
+You can add or modify users in ./configs/authelia/user-database.yml. Additional instructions are provided in that file.
+
```
The default username/password combination (`test`/`test`) should be made explicit here.

jordigh commented on docker-compose-examples/grist-traefik-oidc-auth/README.md:
```diff
@@ -0,0 +1,24 @@
+
+This is automated for you in `generateSecureSecrets.sh`, which uses Authelia's docker image to populate the `./secrets` directory.
+
+This example can then be run with `docker compose up`. This will make Grist available on `https://grist.localhost` with a self-signed certificate (by default), after all the services have started. Note: it may take up to a minute for all of the services to start correctly.
```
Should mention that for local hosting, self-signed certs aren't a problem and should be accepted in the web browser. Not everyone knows how to click past a TLS error.

Spoffy replied:
Added some details on this.

jordigh reviewed (commented):
I was able to get the traefik+oidc example working without any hassle after following the instructions. I'm glad that worked.

One general comment: several files are missing ending newlines.

![image](https://github.com/user-attachments/assets/92c1d4a2-5097-47b3-b9b8-9d88567c5e38)

According to POSIX, every text file should have a final newline.

jordigh commented:
A couple of extra things:

* The secrets creation script modifies a file checked into git and adds other files that show up as untracked. I recommend we use a template `.env` file and have the script copy the template over, and some `.gitignore` rules for the new files it adds.
* Following @paulfitz's preference to avoid more top-level clutter, I think this belongs under `documentation/`, as this is supposed to be almost completely working real examples.

Spoffy commented on docker-compose-examples/grist-traefik-oidc-auth/configs/authelia/configuration.yml:
```diff
@@ -0,0 +1,1414 @@
+# yamllint disable rule:comments-indentation
```
This was copied from their Github, rather than auto-generated. I could add a link to the original Github at the top?

jordigh replied:
Yeah, that link would be helpful.

Spoffy commented on docker-compose-examples/grist-with-postgres-redis-minio/README.md:
```diff
@@ -0,0 +1,17 @@
+
+## How to run this example
+
+This example can be run with `docker compose up`.
```
Added some advice to update the _PASSWORD vars in the `.env` file.

Spoffy commented on docker-compose-examples/grist-traefik-oidc-auth/docker-compose.yml:
```diff
@@ -0,0 +1,118 @@
+      # Domain Grist is hosted at. Custom variable that's interpolated into the Authelia config
+      APP_DOMAIN: 'grist.localhost'
+      # Where Authelia should redirect to after successful authentication.
+      GRIST_OAUTH_CALLBACK_URL: https://grist.localhost/oauth2/callback
```
Parameterised this and the Authelia URLs.

Spoffy commented:
With regards to top-level clutter @jordigh - I'd be worried about discoverability. I'd expect documentation to just be a manual, not examples, personally.

I'd propose we keep it top level (could rename to just docker-compose).

However, we could compromise and do `documentation/docker-compose-examples` and add a section on it to the README?
